File: examples/AODV/APPLrcvcode.m
function [exectime, data] = APPLrcvcode(seg, data)

global received

switch seg,

 case 1,
  exectime = 0.0002;
 case 2,
  msg = ttTryFetch('AODVRcvBox');
  noww = ttCurrentTime;
  
  disp(['Application in Node#' num2str(data.nodeID) ' receiving data: ' num2str(msg)]);
  
  % Store received data
  received = [received msg];
  
  exectime = -1;
end

File: examples/RUNES_demo/node/APPLrcvcode.cpp
double APPLrcvcode(int seg, void* data) {

  APPLrcv_Data* d = (APPLrcv_Data*) data;

  static DataMsg *msg;

  

  switch (seg) {

  case 1:

    ttFetch("AODVRcvBox");

    

    return 0.0;

  case 2:

    msg = (DataMsg*) ttRetrieve("AODVRcvBox");

    

    return 0.0002;

  case 3:    

//	  mexPrintf("(%.4f)Node#%d receiving data from #%d : %f\n", ttCurrentTime(), d->nodeID, msg->src,msg->data);

    ttAnalogOut(msg->src - 1,msg->data);

    delete msg;

    

    // Store received data

    d->received++;

    return FINISHED; 

  }

  

  return FINISHED; // to supress compilation warnings

}

